sygnal/apnspushkin.py

```diff
@@ -131,8 +137,36 @@ def __init__(self, name, sygnal, config):
             if not self.get_config("topic"):
                 raise PushkinSetupException("You must supply topic.")
 
+        http_proxy_url = None
+        http_proxy_creds = None
+
+        # use the Sygnal global proxy configuration
+        proxycfg = sygnal.config["proxy"]
+
+        if proxycfg.get("enabled", False):
+            http_proxy_url = "http://" + proxycfg.get("address")
+            if http_proxy_url is None:
+                raise PushkinSetupException(
+                    "HTTP Proxy enabled for APNs but no URL specified."
+                )
+            else:
+                logger.info("Using HTTP proxy for APNs")
+                # the HTTP proxy code expects a bytestring
+                if proxycfg.get("username") and proxycfg.get("password"):
+                    logger.info("Using HTTP Proxy Basic Auth")
+                    http_proxy_creds = (proxycfg["username"], proxycfg["password"])
+                else:
+                    logger.info("No HTTP Proxy credentials configured")
+
+        loop = asyncio.get_event_loop()
+        if http_proxy_url:
+            # this overrides the create_connection method to use a HTTP proxy
+            loop = ProxyingEventLoopWrapper(loop, http_proxy_url, http_proxy_creds)
```

Comment:
ProxyingEventLoopWrapper appears to expect a host:port rather than a URL?

(suggests more tests are needed?)